shared/src/main/scala/org/atnos/eff/Eff.scala

```diff
@@ -54,16 +54,22 @@ import Eff._
 sealed trait Eff[R, A] {
 
   def map[B](f: A => B): Eff[R, B] =
-    EffApplicative[R].map(this)(f)
+    EffMonad[R].map(this)(f)
```

Comment:
I'm not sure about this. I think I put applicative for a good reason because `map` in `EffMonad` is implemented with `flatMap` and imposes some sequentiality where it's not necessary (maybe I should have commented that).